File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals , new Comparator<int[]>(){
            
            public int compare(int a[] , int b[]){
                if(a[0] != b[0])return Integer.compare(a[0] , b[0]);
                else return Integer.compare(a[1],b[1]);
            }
        });
        
        ArrayList<int[]> list = new ArrayList<>();
        list.add(intervals[0]);
        for(int i = 1 ; i < intervals.length ; i++){
            int []ar = intervals[i] ;
            int st = ar[0];
            int en = ar[1];
            
            int pre[] = list.get(list.size() - 1);
            int str = pre[0] ;
            int end = pre[1] ;
            
            if(st <= end ){
                pre[0] = Math.min(str ,st);
                pre[1] = Math.max(en , end);
            }else{
                list.add(ar);
            }
        }
        int[] arr[] = new int[list.size()][];
        for(int i = 0 ; i < list.size() ; i++){
            arr[i] = list.get(i);
        }
        return arr ;
    }
}

File: 0056-merge-intervals/0056-merge-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
    intervals.sort((a,b)=> {
        if(a[0] != b[0]) return a[0]-b[0] ;
        return a[1]-b[1] ;
    })
    
    const ans = [] ;
    ans.push(intervals[0]); 
    for(let i = 1 ; i < intervals.length; i++){
        const curr = intervals[i] ;
        const last = ans[ans.length-1] ;
        
        if(last[1] >= curr[0] ){
           last[1] = Math.max(curr[1], last[1]) ;
        }else {
            ans.push(intervals[i]) ;
        }
    }
    
    return ans ;
};